### PR TITLE
[Merged by Bors] - feat: port Topology.Algebra.Order.LeftRightLim

### PR DESCRIPTION
**Pull request by mcdoll**
---

- [x] depends on: #2144

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2144
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Topology/Algebra/Order/LeftRightLim.lean [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+
+open Set Filter
+
+open Topology
+
+section
+
+variable {α β : Type _} [LinearOrder α] [TopologicalSpace β]
+
+/-- Let `f : α → β` be a function from a linear order `α` to a topological_space `β`, and
```
Comment (jcommelin) [inline Mathlib/Topology/Algebra/Order/LeftRightLim.lean:48]:
```suggestion
/-- Let `f : α → β` be a function from a linear order `α` to a topological space `β`, and
```

File: Mathlib/Topology/Algebra/Order/LeftRightLim.lean [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+by using the order topology on `α`. If `a` is isolated to its left or the function has no left
+limit, we use `f a` instead to guarantee a good behavior in most cases. -/
+noncomputable def Function.leftLim (f : α → β) (a : α) : β := by
+  classical
+  haveI : Nonempty β := ⟨f a⟩
+  letI : TopologicalSpace α := Preorder.topology α
+  exact if 𝓝[<] a = ⊥ ∨ ¬∃ y, Tendsto f (𝓝[<] a) (𝓝 y) then f a else limUnder (𝓝[<] a) f
+#align function.left_lim Function.leftLim
+
+/-- Let `f : α → β` be a function from a linear order `α` to a topological_space `β`, and
```
Comment (jcommelin) [inline Mathlib/Topology/Algebra/Order/LeftRightLim.lean:59]:
```suggestion
/-- Let `f : α → β` be a function from a linear order `α` to a topological space `β`, and
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**